Tests/Unit/NetworkServerTests/ModuleConnectionHostTest.cs

```diff
@@ -0,0 +1,256 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoRaWan.Tests.Unit.NetworkServerTests
+{
+    using Bogus;
+    using LoRaWan.NetworkServer;
+    using LoRaWan.NetworkServer.BasicsStation.ModuleConnection;
+    using Microsoft.Azure.Devices.Client.Exceptions;
+    using Microsoft.Azure.Devices.Shared;
+    using Moq;
+    using System;
+    using System.Configuration;
+    using System.Net;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Xunit;
+
+    public class ModuleConnectionHostTest
+    {
+        private readonly Faker faker = new Faker();
+
+        [Fact]
+        public void When_Constructor_Receives_Null_Parameters_Should_Throw()
+        {
+            var networkServerConfiguration = new NetworkServerConfiguration();
+            var classCMessageSender = Mock.Of<IClassCDeviceMessageSender>();
+            var loRaDeviceRegistry = Mock.Of<ILoRaDeviceRegistry>();
+
+            // ASSERT
+            Assert.Throws<ArgumentNullException>(() => new ModuleConnectionHost(null, classCMessageSender, loRaDeviceRegistry));
+            Assert.Throws<ArgumentNullException>(() => new ModuleConnectionHost(networkServerConfiguration, null, loRaDeviceRegistry));
+            Assert.Throws<ArgumentNullException>(() => new ModuleConnectionHost(networkServerConfiguration, classCMessageSender, null));
```

Comment:
Best to assert the parameter names too so we know the expected one caused `ArgumentNullException` to be thrown:

```suggestion
            ArgumentNullException ex;
            ex = Assert.Throws<ArgumentNullException>(() => new ModuleConnectionHost(null, classCMessageSender, loRaDeviceRegistry));
            Assert.Equal("networkServerConfiguration", ex.ParamName);
            ex = Assert.Throws<ArgumentNullException>(() => new ModuleConnectionHost(networkServerConfiguration, null, loRaDeviceRegistry));
            Assert.Equal("defaultClassCDevicesMessageSender", ex.ParamName);
            ex = Assert.Throws<ArgumentNullException>(() => new ModuleConnectionHost(networkServerConfiguration, classCMessageSender, null));
            Assert.Equal("loRaDeviceRegistry", ex.ParamName);
```